client/app/models/step.coffee

```diff
@@ -19,6 +19,11 @@ module.exports = class StepModel extends Backbone.Model
 
         @set 'next', next
 
-    # Encapsulate call to step.submit
-    submit: () ->
-        @step.submit()
+
+    submit: (data) ->
+        # TODO: should handle errors here
+        if @step.validate? and (errors = @step.validate(data))
```

Comment:
Just a thought : maybe `validate` will be asynchronous in the future (checking existing email ?), maybe it will need to return some kind of Promise. But again, let's see later.